sql/analyzer/resolve_subqueries.go

```diff
@@ -175,4 +175,37 @@ func cacheSubqueryAlisesInJoins(ctx *sql.Context, a *Analyzer, n sql.Node, scope
 		}
 		return child, nil
 	})
+	if err != nil {
+		return n, err
+	}
+
+	// If the most primary table in the top level join is a CachedResults, remove it.
+	// We only want to do this if we're at the top of the tree.
+	// TODO: Not a perfect indicator of whether we're at the top of the tree...
+	if scope == nil {
+		selector := func(parent sql.Node, child sql.Node, childNum int) bool {
+			if j, isJoin := parent.(plan.JoinNode); isJoin {
+				if j.JoinType() == plan.JoinTypeRight {
+					return childNum == 1
+				} else {
+					return childNum == 0
+				}
+			} else if j, isIndexedJoin := parent.(*plan.IndexedJoin); isIndexedJoin {
+				if j.JoinType() == plan.JoinTypeRight {
+					return childNum == 1
```

Comment:
This is wrong -- indexed joins always put the primary table as the first table, even if it's a right join